packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -37,11 +42,16 @@ export interface VStatic extends BaseVNode {
     fragment: Element;
 }
 
+export interface VFragment extends BaseVNode, BaseVParent {
+    sel: undefined;
+    type: VNodeType.Fragment;
+}
+
 export interface VText extends BaseVNode {
     type: VNodeType.Text;
     sel: undefined;
     text: string;
-    key: undefined;
+    key: Key | undefined;
```

Comment:
this change adds a key to the delimiting text nodes of a fragment. I found an issue with [this](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/integration-karma/test/template/html-comments/index.spec.js#L32) test case in our suite. 

It was failing because it was patching the text delimiter with a comment content, it happens because the criteria to see if 2 vnodes are the same is  `key` and `sel`, which for comments and text was always the case.

Because within an iteration there can't be a text node (only comments and elements), we could always set the same key to comments and it will do the work. I added the key to text nodes (only the delimiters) to avoid an issue if I'm missing a case in which a regular text node collides with the delimiter.